integrations/ethereum.go

```diff
@@ -0,0 +1,92 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stellar/starbridge/model"
+)
+
+// FetchEthTxByHash returns a model.Transaction
+func FetchEthTxByHash(txHash string) (*model.Transaction, error) {
+	conn, e := ethclient.Dial("https://ropsten.infura.io/v3/a42ad89bbcec4ddca5c9abb60eb4a300")
+	if e != nil {
+		return nil, fmt.Errorf("could not connect to infura testnet: %s", e)
+	}
+
+	ctx := context.Background()
+	ethTxReceipt, e := conn.TransactionReceipt(ctx, common.HexToHash(txHash))
+	if e != nil {
+		return nil, fmt.Errorf("could not fetch transaction receipt '%s' by hash from network: %s", txHash, e)
+	}
+	ethTx, isPending, e := conn.TransactionByHash(ctx, common.HexToHash(txHash))
+	if e != nil {
+		return nil, fmt.Errorf("could not fetch transaction '%s' by hash from network: %s", txHash, e)
+	}
+
+	tx, e := Ethereum2Transaction(ethTxReceipt, ethTx, isPending)
+	if e != nil {
+		return nil, fmt.Errorf("could not convert txToString: %s", e)
+	}
+	return tx, nil
+}
+
+// Ethereum2Transaction makes a model.Transaction from an Ethereum Transaction
+func Ethereum2Transaction(txReceipt *types.Receipt, tx *types.Transaction, isPending bool) (*model.Transaction, error) {
+	fromAddress, e := getFromAddress(tx)
+	if e != nil {
+		return nil, fmt.Errorf("unable to get From address: %s", e)
+	}
+
+	var assetInfo *model.AssetInfo
+	// TODO for now I'm only testing conversions of regular ETH payments to any other address, need this to check the address of the Smart-Contract that we deploy
+	if txReceipt.ContractAddress.Hex() == model.AssetEthereum_ETH.ContractAddress {
+		assetInfo = model.AssetEthereum_ETH
+	} else {
+		return nil, fmt.Errorf("unsupported contract address '%s' on Ethereum", txReceipt.ContractAddress.Hex())
+	}
+
+	return &model.Transaction{
```

Comment:
we basically convert to a common format understood by Starbridge so we are not tightly coupled with Stellar's representation throughout our codebase.